File: src/main/java/io/scriptor/impl/CSRMeta.java
package io.scriptor.impl;

import com.carrotsearch.hppc.ObjectIndexedContainer;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.function.LongConsumer;
import java.util.function.LongSupplier;

public record CSRMeta(
        long mask,
        int base,
        @Nullable LongSupplier get,
        @Nullable LongConsumer set,
        @NotNull ObjectIndexedContainer<LongConsumer> getHooks,
        @NotNull ObjectIndexedContainer<LongConsumer> putHooks
) {
}

File: src/main/java/io/scriptor/impl/CSRMeta.kt
package io.scriptor.impl

import java.util.function.Consumer
import java.util.function.Supplier

data class CSRMeta(
    val mask: ULong,
    val base: Int,
    val get: Supplier<ULong>?,
    val set: Consumer<ULong>?,
    val getHooks: MutableList<Consumer<ULong>>,
    val putHooks: MutableList<Consumer<ULong>>,
)
